src/range_selector.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Created by Martin on 15.01.2018.
+ */
+import * as d3 from 'd3';
+
+export interface IDragSelection {
+  dragEnd(sel: d3.Selection<any>);
+  dragStart();
+  dragging(start, end);
+}
+
+class SelectionRect {
+  startPt: [number, number];
+  endPt: [number, number];
+
+  init(point: [number, number]) {
+    this.startPt = point;
+  }
+
+  end(point: [number, number]) {
+    this.endPt = point;
+  }
+
+  getOrderByX() {
+    let startPt = this.startPt, endPt = this.endPt;
+    if(this.startPt[0] > this.endPt[0]) {
+      endPt = this.startPt;
+      startPt = this.endPt;
+    }
+    return [startPt, endPt];
+  }
+}
+
+export default class TimelineRangeSelector {
+  selectionRect: SelectionRect;
+
+  constructor($node: d3.Selection<any>, private candidates: d3.Selection<any>, private listeners: IDragSelection[]) {
+    this.setup($node);
+    this.selectionRect = new SelectionRect();
+  }
+
+  private getSelectionCandidates(dragStart: [number, number], dragEnd: [number, number], $candidates: d3.Selection<any>) {
+    const isInRange = (element: HTMLElement, startPx: number, endPx: number): boolean => {
+      const leftBounds = element.offsetLeft + 10;
+      const rightBounds = element.offsetLeft + 10 + 25;
```

Comment:
What are the 10 and 25 mean? Create variables or even better extract to constants (e.g., `AppConstants`).